win10/RemoteBlinky/RemoteBlinky/ConnectionPage.xaml.cs

```diff
@@ -0,0 +1,338 @@
+                SetUiEnabled( true );
+                return;
+            }
+
+            //connection properties dictionary, used only for telemetry data
+            var properties = new Dictionary<string, string>();
+
+            //use the selected device to create our communication object
+            switch( ConnectionMethodComboBox.SelectedItem as string )
+            {
+                default:
+                case "Bluetooth":
+
+                    //send telemetry about this connection attempt
+                    properties.Add( "Device_Name", device.Name );
+                    properties.Add( "Device_ID", device.Id );
+                    properties.Add( "Device_Kind", device.Kind.ToString() );
+                    App.Telemetry.TrackEvent( "Bluetooth_Connection_Attempt", properties );
+                    App.Connection = new BluetoothSerial( device );
+                    break;
+
+                case "USB":
+
+                    //send telemetry about this connection attempt
+                    properties.Add( "Device_Name", device.Name );
+                    properties.Add( "Device_ID", device.Id );
+                    properties.Add( "Device_Kind", device.Kind.ToString() );
+                    App.Telemetry.TrackEvent( "USB_Connection_Attempt", properties );
+
+                    App.Connection = new UsbSerial( device );
+                    break;
+
+                case "Network":
+                    string host = NetworkHostNameTextBox.Text;
+                    string port = NetworkPortTextBox.Text;
+                    ushort portnum = 0;
+
+                    if( host == null || port == null )
+                    {
+                        ConnectMessage.Text = "You must enter host and IP.";
+                        return;
+                    }
+
+                    try
+                    {
+                        portnum = Convert.ToUInt16( port );
+                    }
+                    catch( FormatException )
+                    {
+                        ConnectMessage.Text = "You have entered an invalid port number.";
+                        return;
+                    }
+
+                    //send telemetry about this connection attempt
+                    properties.Add( "host", host );
+                    properties.Add( "port", portnum.ToString() );
+                    App.Telemetry.TrackEvent( "Network_Connection_Attempt", properties );
+
+                    App.Connection = new NetworkSerial( new Windows.Networking.HostName( host ), portnum );
+                    break;
+            }
+
+            App.Arduino = new RemoteDevice( App.Connection );
+            App.Arduino.DeviceReady += OnConnectionEstablished;
+            App.Arduino.DeviceConnectionFailed += OnConnectionFailed;
+
+            connectionAttemptStartedTime = DateTime.Now;
+            App.Connection.begin( 115200, SerialConfig.SERIAL_8N1 );
```

Comment:
It might be better to create a constant for the baud rate and define it at the top of the class where it's more visible.  Also you might want to add different baud rate constants depending on the connection type.  If 115200 is most common for Bluetooth set it in the ConnectButtonClick function for Bluetooth, if 57600 is most common for USB connections, set that value in ConnectButtonClick.  The default value in the current firmata firmware in Arduino 1.6.5 is 57600.